.circleci/prepare-system.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+# vim: sw=2 et
+set -euo pipefail
+
+fail() {
+  echo "ERROR: $*" >&2
+  exit 1
+}
+
+case "$(uname)" in
+  Linux)
+    if [[ -n ${TARGET:-} ]]; then
+      if [[ $TARGET = FreeBSD ]]; then
+        # cross-compiling to FreeBSD
+        add-apt-repository -y ppa:hvr/ghc
+        apt-get update -qq
+        apt-get install -qy ghc-8.0.2 cabal-install  alex happy ncurses-dev git openssh-client make automake autoconf gcc perl python3 texinfo xz-utils
```

Comment:
Note that `cabal-install` does not exist in my ppa, instead it will result in a likely quite old `cabal` to be installed; my PPA provides `cabal-install` packages suffixed by the major version, e.g. `cabal-install-2.0` or `cabal-install-1.24`, or even `cabal-install-head`.

Reply:
@hvr Thanks for pointing that out. Which version would you suggest to use here with GHC 8.0.2?

Reply:
> GHC 8.0 only works with cabal-install version 1.24 or later. Please upgrade if you have an older version of cabal-install. 

We'll go with 1.24 for now.